jiant/tasks/evaluate/core.py

```diff
@@ -83,6 +83,83 @@ def get_accumulated(self):
         return all_loss
 
 
+class ConcatnateStringListAccumulator(BaseAccumulator):
+    def __init__(self):
+        self.str_list = []
+
+    def update(self, batch_logits, batch_loss, batch, batch_metadata):
+        _, span_pred = batch_logits.max(dim=1)
+        pred_token_start, pred_token_end = batch_logits[:, 0], batch_logits[:, 1]
+        pred_char_start = batch.token_idx_to_char_idx_start[pred_token_start]
+        pred_char_end = batch.token_idx_to_char_idx_end[pred_token_end]
+        self.str_list.extend(
+            [
+                s[i1, i2 + 1]
+                for i1, i2, s in zip(pred_char_start, pred_char_end, batch.selection_str)
+            ]
+        )
+
+    def get_accumulated(self):
+        return self.str_list
+
+
+class SpanPredictionF1andEMScheme(BaseEvaluationScheme):
+    def get_accumulator(self):
+        return ConcatnateStringListAccumulator()
+
+    def get_labels_from_cache_and_examples(self, task, cache, examples):
+        return [datum["data_row"].gt_span_str for datum in cache.iter_all()]
+
+    def get_preds_from_accumulator(self, task, accumulator):
+        return accumulator.get_accumulated()
+
+    def compute_metrics_from_preds_and_labels(self, preds, labels):
+        def normalize_answer(s):
+            """Lower text and remove punctuation, articles and extra whitespace.
+            From official ReCoRD eval script """
```

Comment:
Thanks for leaving this note about ReCoRD. Is this function (and `f1_score()` and `exact_match_score()`) the same as the implementation for [ReCoRD](https://github.com/jiant-dev/jiant/blob/master/jiant/tasks/evaluate/core.py#L373-L415)? Is there anything stopping you from sharing these?

Reply:
> @HaokunLiu Will you have the bandwidth to address @pyeres's suggestions/concerns?

I'm working on matching the performance, will address Phils suggestions after that.